src/library_syscall.js

```diff
@@ -1909,6 +1890,16 @@ for (var x in SyscallsLibrary) {
   var t = SyscallsLibrary[x];
   if (typeof t === 'string') continue;
   t = t.toString();
+  // If a syscall uses FS, but !SYSCALLS_REQUIRE_FILESYSTEM, then the user
+  // has disabled the filesystem or we have proven some other way that this will
+  // not be called in practice, and do not need that code.
+  if (!SYSCALLS_REQUIRE_FILESYSTEM && t.indexOf('FS.') >= 0) {
```

Comment:
Is `t` the raw text of the JS function here?   Will this be too conservative in some cases?   What if I have `myVSF.foo = 1` in my function?  I guess we don't care too much since its local to this file which we control?

Reply:
Yes, `t` is the raw text.

Yes, in theory we should parse this as an AST - but as you said, this is a file we control, so the risk is small.